.github/workflows/reusable_test.yml

```diff
@@ -0,0 +1,65 @@
+name: Reusable Test Workflow
+
+on:
+  workflow_dispatch:
+    inputs:
+      pinned_botorch:
+        required: true
+        type: boolean
+      minimal_dependencies:
+        required: false
+        type: boolean
+        default: false
+  workflow_call:
+    inputs:
+      pinned_botorch:
+        required: true
+        type: boolean
+      minimal_dependencies:
+        required: false
+        type: boolean
+        default: false
+
+jobs:
+  tests-and-coverage:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.9", "3.10"]
+      fail-fast: false
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - if: ${{ inputs.pinned_botorch }}
+      name: Install dependencies with pinned BoTorch (minimal dependencies ${{ inputs.minimal_dependencies }})
+      run: |
+        pip install -e ${{ ((inputs.minimal_dependencies) && '.[unittest_minimal]') || '.[unittest]' }}
```

Comment:
```suggestion
        # The brackets returns '.[unittest_minimal]' if using minimal dependencies and '.[unittest]'
        # otherwise. This saves us from needing 4 install dependencies blocks by supporting two
        # different installation options in one line.
        pip install -e ${{ ((inputs.minimal_dependencies) && '.[unittest_minimal]') || '.[unittest]' }}
```